### PR TITLE
MERC-75 Faceted Search Up to 500 Facets

### PR DESCRIPTION
**Pull request by mickr**
MERC-75 Faceted Search Up to 500 Facets

* Add template to be rendered in modal for search, brand, category
* Refactor javascript for show more functionality to rely on `has_more_results` to show the show more toggle which will fire the modal for all facet results.

### CODE REVIEW

**Comment by mickr** [comment]
Relies on https://github.com/bigcommerce/bigcommerce/pull/13272

**Comment by mickr** [comment]
screen shot 2016-04-28 at 11 30 08 am
screen shot 2016-04-28 at 11 30 28 am

**Comment by mickr** [comment]
kmi2wykesu

**Comment by mickr** [comment]
Wil fix tests

**Comment by mickr** [comment]
@bc-ranji

File: assets/js/theme/common/faceted-search.js [hunk 1]
```diff
@@ -183,6 +167,46 @@ class FacetedSearch {
...
+        const facetUrl = History.getState().url;
+
+        if (this.requestOptions.showMore) {
+            api.getPage(facetUrl, {
+                template: this.requestOptions.showMore,
+                params: {
+                    list_all: facet,
+                },
+            }, (err, response) => {
+                if (err) {
```
Comment (hegrec) [inline assets/js/theme/common/faceted-search.js]: Maybe handle this more appropriately with a user facing error message?

File: assets/js/theme/category.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ export default class Category extends CatalogPage {
                 productListing: 'category/product-listing',
                 sidebar: 'category/sidebar',
             },
+            showMore: 'category/show-more',
```
Comment (hegrec) [inline assets/js/theme/category.js]: What is `category/show-more`?

Comment (mickr) [inline assets/js/theme/category.js]: I forgot to add the templates to git, they are uploaded now.

File: templates/components/faceted-search/show-more-facets.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<div class="modal-body">
+    <h1 class="facet-quick-heading">
+        {{all_facets}}
+    </h1>
+    <div class="form-field" id="facetedSearch-filterItems">
+        <input class="form-input" type="search" placeholder="Find...">
```
Comment (hegrec) [inline templates/components/faceted-search/show-more-facets.html]: should use lang helper

Comment (mickr) [inline templates/components/faceted-search/show-more-facets.html]: Ah good catch

**Comment by bc-ranji** [comment]
💚

**Comment by mickr** [comment]
Anything else here before I merge?

**Pull request merged**